Update the EFS access point configuration to resolve permissions issues

mcdonnnj opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

This pull request updates the [AWS EFS access point](https://docs.aws.amazon.com/efs/latest/ug/efs-access-points.html) configuration to use a specified path in the EFS volume. This includes creating that path if it does not exist and setting expected permissions to mirror the permissions enforced by the access point.
<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

## 💭 Motivation and context ##

A fresh EFS volume is owned by `root:root` and this can only be changed (that I've been able to find) by mounting the EFS without the access point and `chown`ing the mount point. This creates permissions errors because the access point coerces access to the mount point as a specific, non-root UID:GID. We can resolve this by creating a sub-directory in the EFS share that is mounted for the access point and configure that sub-directory to be created with permissions that match the access point configuration.
<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such -->
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## 🧪 Testing ##

Automated tests pass. I confirmed that on a fresh deployment the `/share` mount has the correct permissions. I manually mounted the root EFS volume and confirmed that a directory was created for the access point.
<!-- How did you test your changes? How could someone else test this PR? -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

<!--
## 📷 Screenshots (if appropriate) ##

Uncomment this section if a screenshot is needed.

-->

## ✅ Pre-approval checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

mcdonnnj committed:
Update the EFS access point configuration

Add the root_directory argument to the EFS access point configuration
with settings to ensure that if no root directory exists, then the
created root directory will match the UID/GID settings of the
posix_user argument. We need to specify a non-root path in the EFS to
serve as the root directory for the access point so we can ensure that
permissions are corrected when used to mount the share locally. If we
do not specify a non-root path the root of the EFS will be used, and it
has root:root permissions by default.

jsf9k commented on efs.tf:
```diff
@@ -43,6 +43,18 @@ resource "aws_efs_access_point" "access_point" {
+
+  # Since we only use an access point to access the EFS share we need to make sure
+  # the automatically created root directory (on the EFS share) has the correct
+  # permissions or they will override the local configuration.
```
```suggestion
  # permissions.
```

mcdonnnj replied:
Do you have any thoughts on tweaking that vs outright removal? I'd like to keep mention that the local mount point's permissions will be overridden, which is a problem if they don't align.

jsf9k replied:
I'm fine if it stays, but I find the "override the local configuration" language unclear.  I had to think about it to figure out what you were saying.  "...override the permissions on the local mount point" is a better substitute, so I'd be OK with using that language instead.

In any case, you can disregard if you disagree.  It isn't worth holding up the PR for this tiny point.

mcdonnnj replied:
@jsf9k Would you look at abb95cd and let me know what you think?

jsf9k replied:
Looks good!

jsf9k reviewed (approved):
I made one small suggestion that you can take or leave, but this looks good to me.

dav3r reviewed (approved):
Excellent job figuring this out and fixing it! 🥇 

Someone could still be a dummy and set `efs_access_point_root_directory` to `"/"`, but unless we can think of a simple way to avoid that, I don't think we need to worry too much about this.

mcdonnnj committed:
Update EFS access point root directory comment

Update the comment that explains the needs of the root_directory
argument for the EFS access point used for the assessment data share.
This will hopefully make it clearer why things are configured the way
they are.

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>
